server-spi-private/src/main/java/org/keycloak/validate/builtin/LengthValidator.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.validate.builtin;
+
+import org.keycloak.validate.CompactValidator;
+import org.keycloak.validate.ValidationContext;
+import org.keycloak.validate.ValidationError;
+import org.keycloak.validate.ValidationResult;
+import org.keycloak.validate.ValidatorConfig;
+
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class LengthValidator implements CompactValidator {
+
+    public static final LengthValidator INSTANCE = new LengthValidator();
+
+    public static final String ID = "length";
+
+    public static final String MESSAGE_INVALID_LENGTH = "error-invalid-length";
+    public static final String KEY_MIN = "min";
+    public static final String KEY_MAX = "max";
+
+    private static final ValidatorConfig DEFAULT_CONFIG;
+
+    static {
+        Map<String, Object> config = new HashMap<>();
+        config.put(KEY_MIN, 0);
+        config.put(KEY_MAX, 255);
+
+        DEFAULT_CONFIG = ValidatorConfig.configFromMap(config);
+    }
+
+    private LengthValidator() {
+    }
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    @Override
+    public ValidationContext validate(Object input, String inputHint, ValidationContext context, ValidatorConfig config) {
+
+        if (input == null) {
+            context.addError(new ValidationError(ID, inputHint, MESSAGE_INVALID_LENGTH, input));
+            return context;
+        }
```

Comment:
I noticed in general some inconsistencies in the built-in validators in how the validation is performed if input is null/empty. I think we might want to skip validations by default if given input is null/empty. Non-empty validation should be probably done explicitly if needed. Or maybe this could be somehow configurable in the context?

Reply:
I agree, the current implementations are WIP and needs to be aligned. I like your idea of handling null values.

Reply:
For user profile, we need all these "format" validations to run only if value is not null nor empty